homeassistant/components/stream/core.py

```diff
@@ -177,7 +159,7 @@ def cleanup(self):
         """Handle cleanup."""
         self._event.set()
         self._idle_timer.clear()
-        self._segments = deque(maxlen=MAX_SEGMENTS)
+        self._segments = deque(maxlen=self._segments.maxlen)
```

Comment:
Would `clear()` work just as well?

Reply:
The `recorder_save_worker` might still be running on the original deque so `clear()` might break that